request/real_ip.go

```diff
@@ -22,14 +25,38 @@ func RealIP(r *http.Request) string {
 	}
 
 	if fw := r.Header.Get(header.XForwardFor); fw != "" {
-		// X-Forwarded-For has no port
-		if i := strings.IndexByte(fw, ','); i >= 0 {
-			fw = fw[:i]
+		xffs := strings.Split(fw, ",")
+
+		// If no IPs, return the first IP in the chain
+		if len(xffs) == 0 {
+			return ""
+		}
+
+		// Get depth from config, default to 0 (first IP in chain)
+		depth := 0
+		if Global != nil {
+			depth = Global().HttpServerOptions.XFFDepth
+		}
+
+		// If depth exceeds available IPs, return empty
+		if depth > len(xffs) {
+			return ""
+		}
+
+		// Choose the appropriate IP based on depth
+		// depth=0 means first IP (leftmost), depth=1 means last IP, depth=2 means second to last, etc.
+		var ip string
+		if depth == 0 {
```

Comment:
I think this should be `if depth <= 0` to avoid `index out of bounds` panics in the else case